pkg/artifactory/resource/configuration/resource_artifactory_mail_server.go

```diff
@@ -0,0 +1,338 @@
+	r.Enabled = types.BoolValue(mailServer.Enabled)
+	r.ArtifactoryURL = types.StringValue(mailServer.ArtifactoryURL)
+	r.From = types.StringValue(mailServer.From)
+	r.Host = types.StringValue(mailServer.Host)
+	r.Username = types.StringValue(mailServer.Username)
+	r.Port = types.Int64Value(mailServer.Port)
+	r.SubjectPrefix = types.StringValue(mailServer.SubjectPrefix)
+	r.UseSSL = types.BoolValue(mailServer.UseSSL)
+	r.UseTLS = types.BoolValue(mailServer.UseTLS)
+
+	return nil
+}
+
+func NewMailServerResource() resource.Resource {
+	return &MailServerResource{}
+}
+
+type MailServerResource struct {
+	ProviderData utilsdk.ProvderMetadata
+}
+
+func (r *MailServerResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "artifactory_mail_server"
+}
+
+func (r *MailServerResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides an Artifactory mail server config resource. This resource configuration corresponds to mail server config block in system configuration XML (REST endpoint: artifactory/api/system/configuration). Manages mail server settings of the Artifactory instance.",
+		Attributes: map[string]schema.Attribute{
+			"enabled": schema.BoolAttribute{
+				MarkdownDescription: "When set, mail notifications are enabled.",
+				Required:            true,
+			},
+			"artifactory_url": schema.StringAttribute{
+				MarkdownDescription: "The Artifactory URL to to link to in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsURLHttpOrHttps(),
+				},
+			},
+			"from": schema.StringAttribute{
+				MarkdownDescription: "The 'from' address header to use in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsEmail(),
+				},
+			},
+			"host": schema.StringAttribute{
+				MarkdownDescription: "The mail server IP address / DNS.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The username for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"password": schema.StringAttribute{
+				MarkdownDescription: "The password for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"port": schema.Int64Attribute{
```

Comment:
Technically, it's an `int16` since port can never be greater than 65535 (2^16 -1). Validate in range?

Reply:
There's no `Int16Attribute` type. I will put a validator on this.